mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerProvider.java

```diff
@@ -53,11 +56,26 @@
 import org.apache.james.metrics.tests.RecordingMetricFactory;
 
 import com.datastax.driver.core.Session;
+import com.google.inject.Module;
 
 public class CassandraMailboxManagerProvider {
     private static final int LIMIT_ANNOTATIONS = 3;
     private static final int LIMIT_ANNOTATION_SIZE = 30;
 
+    static CassandraMailboxManager provideMailboxManager(Session session, CassandraTypesProvider cassandraTypesProvider,
+                                                         PreDeletionHooks preDeletionHooks, Module... overriedGuiceModules) {
```

Comment:
Yeah, it's suck, but I have to compromise to not get this PR being big. Otherwise, I fuck up my estimate :(

Reply:
It's an estimate, not a commitment, doing that refactoring early (maybe in another PR), will avoid confusion and velocity decrease.

Reply:
Estimate should not impact on your way to solve an issue. There is a simple reason for that: if you overestimated a task, and it took you 1 day instead of 3 days, will you still work on this task during 2 days to compensate?
By the way that's why we estimate in points and not in time. We evaluate the complexity of the task, not the time it will take us to do it.

Reply:
We could introduce a simple `builder` here...

For the phylosophical debate here, my take on it is that there are several issues composing a sprint, some will take longer, some less.

Reply:
> We could introduce a simple `builder` here...

To be more explicit, I was not arguing here to force you to add a builder or not. I was arguing  because the reason invoked by Duc is not valid for me, and I wanted to explain it and make it clear why.

> For the phylosophical debate here, my take on it is that there are several issues composing a sprint, some will take longer, some less.

Yes, it's an other simpler way to say that.

Reply:
> Estimate should not impact on your way to solve an issue. There is a simple reason for that: if you overestimated a task, and it took you 1 day instead of 3 days, will you still work on this task during 2 days to compensate?

Here my task is underestimated and I'm trying to minimize the change set.